libs/live-signer-solana/src/DMKSignerSolana.ts

```diff
@@ -0,0 +1,179 @@
+    console.log("ERROR", error);
+    if (
+      typeof error.originalError !== "object" ||
+      error.originalError === null ||
+      !("errorCode" in error.originalError)
+    ) {
+      return new Error(error._tag);
+    }
+    if (
+      typeof error.originalError === "object" &&
+      error.originalError !== null &&
+      "errorCode" in error.originalError &&
+      typeof error.originalError.errorCode === "string"
+    ) {
+      switch (error.originalError.errorCode) {
+        case "6985":
+          return new UserRefusedOnDevice();
+        default:
+          return new Error(error._tag);
+      }
+    } else {
+      return new Error(error._tag);
+    }
+  }
+
+  /**
+   * fetches current app configuration from the device via DMK
+   */
+  async getAppConfiguration(): Promise<AppConfig> {
+    const { observable } = this.dmkSigner.getAppConfiguration();
+    return new Promise<AppConfig>((resolve, reject) => {
+      observable.subscribe({
+        next: state => {
+          if (state.status === DeviceActionStatus.Error) {
+            reject(this._mapError<GetAppConfigurationDAError>(state.error));
+          }
+          if (state.status === DeviceActionStatus.Completed) {
+            const { version, blindSigningEnabled, pubKeyDisplayMode } = state.output;
+            const mode =
+              pubKeyDisplayMode === this.DMKPubKeyDisplayMode.long
+                ? PubKeyDisplayMode.LONG
+                : PubKeyDisplayMode.SHORT;
+            resolve({ version, blindSigningEnabled, pubKeyDisplayMode: mode });
+          }
+        },
+        error: err => {
+          reject(err);
+        },
+      });
+    });
+  }
+
+  /**
+   * retrieves a Solana address for the given derivation path.
+   * @param path - BIP32 derivation path
+   * @param display - whether to prompt user confirmation on device
+   */
+  async getAddress(path: string, display?: boolean): Promise<SolanaAddress> {
+    console.log("Fetching address for path via DMK");
+    const { observable } = this.dmkSigner.getAddress(path, { checkOnDevice: !!display });
+    return new Promise<SolanaAddress>((resolve, reject) => {
+      observable.subscribe({
+        next: state => {
+          if (state.status === DeviceActionStatus.Error) {
+            reject(this._mapError<GetAddressDAError>(state.error));
+          }
+          if (state.status === DeviceActionStatus.Completed) {
+            const addressString = state.output;
+            const addressBytes = Buffer.from(bs58.decode(addressString));
+            resolve({ address: addressBytes });
+          }
+        },
+        error: err => {
+          reject(err);
+        },
+      });
+    });
+  }
+
+  /**
+   * signs a Solana transaction via DMK.
+   * @param path - BIP32 derivation path
+   * @param txBuffer - transaction data as a uint8 array
+   */
+  async signTransaction(path: string, txBuffer: Uint8Array): Promise<SolanaSignature> {
+    console.log("Signing transaction via DMK");
```

Comment:
We probably want to remove this console.log

Reply:
Removed